Route/CmsPageRouter.php

```diff
@@ -207,7 +210,26 @@ protected function generateFromPage(PageInterface $page, array $parameters = arr
             throw new \RuntimeException(sprintf('Page "%d" has no url or customUrl.', $page->getId()));
         }
 
-        return $this->decorateUrl($url, $parameters, $referenceType);
+        // NEXT_MAJOR: remove this if block
+        if (!$this->context instanceof SiteRequestContext) {
+            @trigger_error(
+                sprintf('Since, 3.x, when calling %s, %s::$context should implement SiteRequestContextInterface. This will become mandatory in 4.x', __METHOD__, __CLASS__),
```

Comment:
I'm turning a blind eye on what I think is a linebreaking issue until [this](https://github.com/sonata-project/dev-kit/issues/218) is resolved